app/javascript/dashboard/components-next/sidebar/Sidebar.vue

```diff
@@ -151,6 +155,7 @@ const menuItems = computed(() => {
       icon: 'i-lucide-inbox',
       to: accountScopedRoute('inbox_view'),
       activeOn: ['inbox_view', 'inbox_view_conversation'],
+      showBadge: hasUnreadNotifications.value || false,
```

Comment:
@iamsivin let's do this. a bit differently. Instead of passing a boolean, let's just pass the key to it

```js
getterKeys: {
  counter: 'notifications/unreadCount'
  badge: 'notifications/hasUnreadNotifications'
}
```

We can then use this in the component using `useMapGetter` directly

```js
const showBadge = useMapGetter(props.getterKeys.badge);
const counter = useMapGetter(props.getterKeys.counter);
const showCounter = computed(() => {
  return counter.value > 0
})
```

This will keep the logic for these from bleeding into the `Sidebar` component. And will be make it reactive at a much higher granularity (This whole Sidebar will not be updated, only the SidebarLeaf node will) when something changes.

Reply:
We don't have to implement the `counter` part for now, that is for the future cases, just implement the badge part for now

Reply:
@scmmishra Done